ESQL: Update docs for new types

alex-spies opened the pull request:
`aggregate_metric_double` and `dense_vector` are a bit special on 9.2.0, but not anymore on 9.2.1. Let's make that clear and let's add their corresponding type conversion functions (`to_...`) to our public docs.

alex-spies committed:
Update docs for new types

aggregate_metric_double and dense_vector

elasticsearchmachine commented:
Pinging @elastic/es-analytical-engine (Team:Analytics)

elasticsearchmachine commented:
Pinging @elastic/core-docs (Team:Docs)

alex-spies committed:
Fix typos

alex-spies committed:
Try to fix applies_to markers

kkrik-es reviewed (approved):
Thanks Alex, @limotova fyi.

alex-spies committed:
Fixes

leemthompo commented on docs/reference/query-languages/esql/_snippets/lists/type-conversion-functions.md:
```diff
@@ -1,10 +1,12 @@
 * [`TO_DATETIME`](../../functions-operators/type-conversion-functions.md#esql-to_datetime)
 * [`TO_DATE_NANOS`](../../functions-operators/type-conversion-functions.md#esql-to_date_nanos)
 * [`TO_DEGREES`](../../functions-operators/type-conversion-functions.md#esql-to_degrees)
+* [`TO_DENSE_VECTOR`](../../functions-operators/type-conversion-functions.md#esql-to_dense_vector)
```
```suggestion
* [`TO_DENSE_VECTOR`](../../functions-operators/type-conversion-functions.md#esql-to_dense_vector) {applies_to} `stack: preview 9.2.0` {applies_to}`serverless: preview`
```

alex-spies committed:
Mark conversion functions as preview

leemthompo commented on docs/reference/query-languages/esql/limitations.md:
```diff
@@ -106,9 +107,10 @@ Some [field types](/reference/elasticsearch/mapping-reference/field-data-types.m
 
-* `dense_vector` field type is partially supported. [`KNN` function](elasticsearch://reference/query-languages/esql/functions-operators/dense-vector-functions.md#esql-knn) queries will work and any field data will be retrieved as part of the results. However, the type will appear as `unsupported` when the `KNN` function is not used.
+* {applies_to}`stack: preview 9.2.0` `dense_vector` field type is only partially supported on this patch version (fixed in {applies_to}`stack: preview 9.2.1`). [`KNN` function](elasticsearch://reference/query-languages/esql/functions-operators/dense-vector-functions.md#esql-knn) queries will work and any field data will be retrieved as part of the results. However, the type will appear as `unsupp ...
+* {applies_to}`stack: preview 9.2.0` `aggregate_metric_double` field type is only partially supported on this patch version (fixed in {applies_to}`stack: preview 9.2.1`). Using the [`TO_AGGREGATE_METRIC_DOUBLE` function](elasticsearch://reference/query-languages/esql/functions-operators/type-conversion-functions#esql-to_aggregate_metric_double) will work and any field data will be retrieved as pa ...
```
```suggestion
* {applies_to}`stack: preview 9.2.0` `aggregate_metric_double` field type is only partially supported on this patch version (fixed in {applies_to}`stack: preview 9.2.1`). Using the [`TO_AGGREGATE_METRIC_DOUBLE` function](elasticsearch://reference/query-languages/esql/functions-operators/type-conversion-functions.md#esql-to_aggregate_metric_double) will work and any field data will be retrieved as part of the results. However, the type will appear as `unsupported` if this function is not used.
```

alex-spies committed:
Use consistent links

alex-spies commented:
@limotova , I noticed that our generated function docs don't mention the preview status of `TO_AGGREGATE_METRIC_DOUBLE`, I think we need to follow up on this:
<img width="770" height="159" alt="image" src="https://github.com/user-attachments/assets/bfddec26-474a-404c-8141-660cb7e873a0" />
It should probably look similar to this:
<img width="770" height="159" alt="image" src="https://github.com/user-attachments/assets/f276ddad-6f75-4675-b59c-b90103b078c4" />

alex-spies committed:
Mention TO_DENSE_VECTOR

leemthompo committed:
Refactor 9.2.0 type limitations and fix cross-repo links

Grouped dense_vector and aggregate_metric_double limitations under single bullet with sub-bullets
Added subheading "Partial support in 9.2.0"
Fixed broken elasticsearch:// cross-repo links to /reference/ format
Converted multi-index paragraph to bullet point
Wrapped explanation in note block with bulleted examples

carlosdelest reviewed (approved):
LGTM, thanks @alex-spies !

limotova reviewed (approved):
Thank you!

alex-spies commented:
Thank you @leemthompo for fostering this into main!
